### PR TITLE
Feature/adding yup validation

### PR DESCRIPTION
**Pull request by mark-escosura**
## Description

I used the yup library to validate InstructorAddCourseCard with @seanmp3 and @19jlevitre 

Created a File: InstructorAddCourseFormSchema and added a schema for size, min_age, max_age, start_time, end_time, sessions, and location when they are typed in and then left blank. Unfortunately, we couldn't figure out how to validate when start_date is after end_date and vice versa. The same goes for age, and time.

Fixes # (issue)

## Loom Video

(Draft Pull Request)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Comment by mark-escosura** [comment]
[Add validation to instructor create class form](https://trello.com/c/w1N9mgO1/166-add-validation-to-instructor-create-class-form)

**Comment by natecheney-dev** [comment]
Great job in implementing yup validation while also cleaning up the previous code and changing it to fit what we need.

**Review by natecheney-dev — APPROVED** [review]
Great job in implementing yup validation while also cleaning up the previous code and changing it to fit what we need.

**Review by ElchinNasirov — APPROVED** [review]

File: src/components/pages/InstructorAddCourse/InstructorAddCourseFormSchema.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  min_age: yup
+    .number()
+    .lessThan(18, 'Must be 17 years or younger')
+    .moreThan(6, 'Must be 7 years or older')
+    .required('You must specify a minimum age'),
+  max_age: yup
+    .number()
+    .lessThan(18, 'Must be 17 years or younger')
+    .moreThan(6, `Must be 7 years or older`)
+    // .moreThan(min_age, `Must be older than ${min_age}`)
```
Comment (PriscilaMonteiro) [inline src/components/pages/InstructorAddCourse/InstructorAddCourseFormSchema.js:31]: Is this comment necessary?

**Review by PriscilaMonteiro — COMMENTED** [review]
Thanks Mark to start this Draft.  Yup is an amazing tool that validates the shape of an existing value. We needed this in our project.

**Comment by ModupeD** [comment]
Thanks for starting this. We still need to validate it, but you did a good job.

File: src/components/pages/InstructorAddCourse/InstructorAddCourseFormSchema.js (hunk 1 above)
Comment (mark-escosura) [reply src/components/pages/InstructorAddCourse/InstructorAddCourseFormSchema.js]: Yeah i will take this one out. I just had it as a reference

File: src/components/pages/InstructorAddCourse/InstructorAddCourseCards.js [hunk 2]
```diff
@@ -32,13 +37,41 @@ const initialClassDataState = {
...
+  sessions: '',
+  location: '',
+  open_seats_remaining: '',
+  instructor_id: '',
+};
+
 const InstructorAddCourseCards = props => {
   const [classData, setClassData] = useState(initialClassDataState);
+  const [formErrors, setFormErrors] = useState(initialClassDataStateFormErrors);
+  const [disabled, setDisabled] = useState(true);
```
[the remark this answers is not in the document]
Comment (mark-escosura) [reply src/components/pages/InstructorAddCourse/InstructorAddCourseCards.js:59]: I am going to get rid of this warning on my next commit

File: src/components/pages/InstructorAddCourse/InstructorAddCourseCards.js [hunk 3]
```diff
@@ -64,25 +99,22 @@ const InstructorAddCourseCards = props => {
         <Form
           wrapperCol={{ span: 50 }}
           layout="horizontal"
+          className="iadc__form"
```
Comment (ryansnorek) [inline src/components/pages/InstructorAddCourse/InstructorAddCourseCards.js:102]: Is this class name supposed to have 2 underscores?

**Review by ryansnorek — COMMENTED** [review]
Missing video recording but the code looks good. Wasn't sure about the double underscore class name so I left a comment

**Pull request merged**